### ISSUE TITLE
How to assign young Modulus for material in mfem example 2?

### ISSUE THREAD

**Issue by NikhilPurbey** [mfem/mfem#4536]
Hello there, 

      How can I assign young modulus to mfem example 2 material.

 I have a seperate ex2.cpp which I have modified to do linear elasticity analysis for a single material beam. Here is my ex2.cpp 
[ex2.zip](https://github.com/user-attachments/files/17331700/ex2.zip)

I am using this msh file with my ex2.cpp 
[newRefinedTestBeam.zip](https://github.com/user-attachments/files/17331718/newRefinedTestBeam.zip)

Also, I wanted to clarify how I could assign a value pull force so that basically assign weight of the material and it does the linear elasticity calculation. in my ex2.cpp I wanted to be able to assign weight of beam as pullforce array value, and be able to assign young modulus depending on the type of material I am analysing.

How can I accomplish that? Maybe the way I am trying to do this is wrong. If I could get some more guidance on it it will be super helpful.

**Comment by NikhilPurbey** [comment]
@Raghavendra0117 Do you have any insight on this?

**Comment by Raghavendra0117** [comment]
Hi @NikhilPurbey , i am not very sure about the following answer but worth trying.

λ and μ are the material Lame constants and can be evaluated using the following formulae.

$$ \lambda = \frac {E \nu}{\left(1- 2 \nu \right) \left(1+ \nu \right)}  $$

$$ \mu = \frac{E}{2 \left( 1+ \nu \right)} $$

E = Elasticity modulus
μ = poisson ratio.

Ref:[https://mfem.org/examples/](url).
[https://en.wikipedia.org/wiki/Lam%C3%A9_parameters](url)

**Comment by NikhilPurbey** [comment]
Hi @Raghavendra0117,

   I think you meant μ means Shear Modulus. But that was really helpful. Thanks! 

I need some guidance. The results seems to be different than the calculations I actually make. This is the calculation I did.

For example, lets assume the mesh is 10m X 1m X 1m steel bar. Density of the steel is 7850kg/m^3. Young Modulus (E) = 200 * 10^9 GPa. Moment of Area is 1/12 because cross section is 1m * 1m. If I calculate the bending of steel bar under its own weight, using the formula for beam fixed between two points, using formula delta = FL^3/(192EI) I get delta as 0.0002404m. 

But if I try to do same thing in the my modified ex2.cpp along with my modified mesh the delta comes as 0.0002668m. I was wondering is this normal error that I should expect? Also to get the value closer to answer I had to input the value of pullforce as Distributed load acting over the whole boundary. for example, The weight of steel bar would be mass * 9.8 = 769300. but if I add this it would give a huge value. So I figured maybe I should converted it into distributed load. Hence I divided the weight by 10 (length of the bar) and it gave me around double the correct answer. I thought maybe this is because as This force is acted on the boundary, hence it is not only acting on the upper surface but lower surface as well. Hence I did 769300/(10*2) and got 38465 as my pull force value. 

The thing is I am not sure if all the assumptions I made are correct, and I am not sure if the difference in my calculation and mfem solution is acceptable, especially in the cases of structural mechanics.

Here I have attached my version of example2.cpp and the mesh file I am using.  
[ex2_and_msh.zip](https://github.com/user-attachments/files/17528287/ex2_and_msh.zip)

Thanks, 
Nikhil
